flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -263,8 +279,20 @@ public Builder uidPrefix(String newPrefix) {
       // Distribute the records from input data stream based on the write.distribution-mode.
       rowDataInput = distributeDataStream(rowDataInput, table.properties(), table.spec(), table.schema(), flinkRowType);
 
+      // Convert the INSERT stream to be an UPSERT stream if needed.
```

Comment:
I think this comment should be wrote as:

```
// Validate the equality fields and partition fields if we enable the upsert stream.
```

Because we did not do any data stream conversion as the stale comment said.